src/main/java/gift/product/repository/JpaProductRepositoryAdapter.java

```diff
@@ -0,0 +1,60 @@
+package gift.product.repository;
+
+import gift.global.common.dto.SortInfo;
+import gift.product.domain.Product;
+import gift.product.exception.ProductNotFoundException;
+import gift.product.jpa.ProductEntity;
+import gift.product.jpa.ProductMapper;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.domain.Sort.Direction;
+
+public class JpaProductRepositoryAdapter implements ProductRepository {
+
+  private final JpaProductRepository jpaProductRepository;
+
+  public JpaProductRepositoryAdapter(JpaProductRepository jpaProductRepository) {
+    this.jpaProductRepository = jpaProductRepository;
+  }
+
+  @Override
+  public Long save(Product product) {
+    ProductEntity savedProductEntity = jpaProductRepository.save(ProductMapper.toEntity(product));
+    return savedProductEntity.getId();
+  }
+
+  @Override
+  public Optional<Product> findById(Long id) {
+    return jpaProductRepository.findById(id).map(ProductMapper::toDomain);
+  }
+
+  @Override
+  public List<Product> findAll() {
+    return jpaProductRepository.findAll().stream()
+        .map(ProductMapper::toDomain)
+        .toList();
+  }
+
+  @Override
+  public List<Product> findAllByPage(int offset, int pageSize, SortInfo sortInfo) {
+    Direction direction = sortInfo.isAscending() ? Direction.ASC : Direction.DESC;
```

Comment:
현재 `정렬`에 대한 옵션으로는 (오름차순인지 여부로) true, false를 받고있는데요. 
애초에 `"asc"`, `"desc"` 와 같이 문자열을 받는다면 아래와 같이 변경할 수 있지 않을까요? 

```suggestion
    Direction direction =  Direction.fromString(sortInfo.sort());
```

혹은 애초에 "ASC", "DESC"로 정렬 요청 값을 받는다면,  아래와 같이 애초에 요청 파라미터를 `Direction` 타입으로 받을 수도 있을 것 같아요. 

```suggestion
     Direction direction = sortInfo.sort();
```

Reply:
Spring data jpa를 사용하지 않고 있을 때 페이지네이션 기능을 구현했다가 점차 리팩토링하면서 애매모호해진 것 같습니다😂
좀 더 명확하게 처음부터 MethodArgumenteResolver에서 Direction을 검증 및 기본값 처리 후 받아도록 리팩토링했습니다!